aiida/cmdline/params/types/path.py

```diff
@@ -31,3 +33,76 @@ def convert(self, value, param, ctx):
 
     def __repr__(self):
         return 'ABSOLUTEPATH'
+
+
+class ImportPath(click.Path):
+    """AiiDA extension of Click's Path-type to include URLs
+    An ImportPath can either be a `click.Path`-type or a URL.
+    """
+
+    # pylint: disable=protected-access
+
+    def __init__(self, timeout_seconds=None, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+        self.timeout_seconds = timeout_seconds
+
+    def convert(self, value, param, ctx):
+        """Overwrite `convert`
+        Check first if `click.Path`-type, then check if URL.
+        """
+        try:
+            # Check if `click.Path`-type
+            return super(ImportPath, self).convert(value, param, ctx)
+        except click.exceptions.BadParameter:
+            # Check if URL
+            return self.checks_url(value, param, ctx)
+
+    def checks_url(self, value, param, ctx):
+        """Do checks for possible URL path"""
+        from six.moves import urllib
+        from django.core.validators import URLValidator
```

Comment:
Do we really need a django dependency here? I know we have it anyway, but if at some point it needs to be dropped. Isn't there a simply python analogue. Or better, do not validate it all, but literally just go to the next block and try to open it and catch the exception. Maybe the `urllib` library already throws  a specific exception for malformed URLs

Reply:
I have moved it to the next block and removed URL validation, other than through `urlopen()`.